projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java

```diff
@@ -5,91 +5,55 @@
 
-import hasadna.noloan2.protobuf.SmsProto.SMSmessage;
+import hasadna.noloan2.protobuf.SMSProto.SmsMessage;
 import noloan.R;
 
-public class MainActivity extends AppCompatActivity implements View.OnClickListener {
-  
-  private static final int PERMISSION_REQUEST_CODE = 123;
+public class MainActivity extends AppCompatActivity {
   
-  EditText name;
-  RecyclerView userView;
+  TextView text;
   
-  FirebaseFirestore firestoreClient;
-  Query query;
+  ArrayList<SmsMessage> spamList;
   
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
     
-    initFirestore();
+    text = findViewById(R.id.TV_text);
     
-    name = findViewById(R.id.ET_name);
+    spamList = new ArrayList<>();
+    ArrayList<SmsMessage> messages = readSms();
     
-    Button add = findViewById(R.id.B_adduser);
-    add.setOnClickListener(this);
+    // For testing, adding the first message read to the list of spam.
+    spamList.add(SmsMessage.newBuilder().setSender(messages.get(0).getSender()).setBody(messages.get(0).getBody()).build());
     
-    userView = findViewById(R.id.RV_userview);
-    userView.setAdapter(new SmsAdapter(readSms()));
-    // Firestore test
-    /*UserAdapter adapter = new UserAdapter(query);
-    userView.setAdapter(adapter);
-    adapter.startListening();*/
-    userView.setLayoutManager(new LinearLayoutManager(this));
-    
-  }
-  
-  private void initFirestore() {
-    firestoreClient = FirebaseFirestore.getInstance();
-    query = firestoreClient.collection("Users");
-  }
-  
-  // Write the user to the Firestore
-  private void writeUser(loguser user) {
-    FirestoreElement userElement = encodeUser(user);
+    int count = 0;
+    for (SmsMessage message : messages) {
+      for (SmsMessage spam : spamList) {
+        if (comparingMessages(message, spam)) {
+          count++;
+          break;
+        }
+      }
+    }
     
-    CollectionReference users = firestoreClient.collection("Users");
-    users.add(userElement);
-  }
-  
-  // Encode user proto to base64 for storing in Firestore
-  private FirestoreElement encodeUser(loguser user) {
-    byte[] protoBytes = user.toByteArray();
-    String base64BinaryString = Base64.encodeToString(protoBytes, Base64.DEFAULT);
-    return new FirestoreElement(base64BinaryString);
+    text.setText(count + "");
   }
   
-  @Override
-  public void onClick(View v) {
-    switch (v.getId()) {
-      case R.id.B_adduser: // Clicking on the add user button
-        loguser user = loguser.newBuilder()
-          .setName(name.getText().toString())
-          .build();
-        writeUser(user);
-    }
+  // Basic comparing of two SMS massages
+  private boolean comparingMessages(SmsMessage m1, SmsMessage m2) {
+    return m1.equals(m2);
   }
   
   //Reads SMS. If no permissions are granted, exit app.
-  private ArrayList<SMSmessage> readSms() {
+  private ArrayList<SmsMessage> readSms() {
```

Comment:
We ask for permissions on app startup, right?
So not having permissions should be an error and let's log it as such.